### PR TITLE
[Sprint: 41] XD-2562 Fixing dependency issues for running tests in STS

### PR DESCRIPTION
**Pull request by trisberg**

### CODE REVIEW

File: gradle/build-hadoop-distros.gradle [hunk 1]
```diff
@@ -28,6 +28,7 @@ project('spring-xd-hadoop:hadoop24') {
 		runtime ("org.apache.hadoop:hadoop-mapreduce-client-jobclient:$hadoop24Version")
 		runtime ("org.apache.hadoop:hadoop-streaming:$hadoop24Version")
 		runtime ("org.apache.hadoop:hadoop-yarn-common:$hadoop24Version")
+		configurations.runtime.exclude(group: 'org.xerial.snappy')
```
Comment (dturanski) [inline gradle/build-hadoop-distros.gradle]: These will not get picked up by the maven plugin. If they need to be excluded from the pom need to specify the modules explicitly

Comment (trisberg) [inline gradle/build-hadoop-distros.gradle]: These aren't "real" projects so not too worried about the pom, really would be better to not generate a pom at all.

Comment (trisberg) [inline gradle/build-hadoop-distros.gradle]: The only reason we need these excludes are for our tests to run in an IDE, so for the pom files we really don't need theese excludes

Comment (dturanski) [inline gradle/build-hadoop-distros.gradle]: I added the specific modules to excludes to fix spring-xd-dirt and spring-xd-test poms which are critical to spring-xd-module-parent  (also spring-xd-module and spring-xd-module-spi). As long as we can still run SingleNodeApplication from an external maven project. Unfortunately there are no automated tests for this.

File: build.gradle [hunk 2]
```diff
@@ -996,27 +996,23 @@ project('spring-xd-test') {
 		compile "org.springframework.integration:spring-integration-amqp"
 		compile "org.springframework.integration:spring-integration-redis"
 		compile "org.springframework.integration:spring-integration-mqtt"
-		compile("org.springframework.data:spring-data-hadoop") {
-			exclude group: 'org.springframework.batch', module: 'spring-batch-core'
```
Comment (dturanski) [inline build.gradle]: You changed a lot of stuff here. Are the original exclusions preserved?

**Comment by dturanski** [comment]
I ran ./gradlew install to build the spring-xd-module-parent pom and then ran the build for a couple of module projects. LGTM.

File: build.gradle (hunk 2 above)
Comment (trisberg) [inline build.gradle]: I think the old excludes were an outdated list - matched the new excludes with what is in the spring-xd-hadoop module minus a few that we need to get the minicluster running for the tests

**Pull request merged**